test/model/test_node.py

```diff
@@ -506,7 +510,77 @@ async def test_value_added_events(multisensor_6):
     )
     node.handle_value_added(event)
     assert isinstance(event.data["value"], ConfigurationValue)
-    assert isinstance(node.values["52-112-0-2"], ConfigurationValue)
+    assert isinstance(node.values[value_id], ConfigurationValue)
+    # ensure that the value was added to the node's state data
+    assert node.value_data_idx(value_id)
+
+
+async def test_value_updated_events(multisensor_6):
+    """Test Node value updated events."""
+    node = multisensor_6
+    value_id = "52-112-0-2"
+    event = Event(
+        type="value updated",
+        data={
+            "source": "node",
+            "event": "value updated",
+            "nodeId": 52,
+            "args": {
+                "commandClassName": "Configuration",
+                "commandClass": 112,
+                "endpoint": 0,
+                "property": 2,
+                "propertyName": "Stay Awake in Battery Mode",
+                "newValue": 1,
+                "prevValue": 0,
+            },
+        },
+    )
+    node.handle_value_updated(event)
+    assert isinstance(event.data["value"], ConfigurationValue)
+    assert isinstance(node.values[value_id], ConfigurationValue)
+    # ensure that the value was added to the node's state data
+    assert (value_idx := node.value_data_idx(value_id))
+    # ensure that the node's state data was updated and that old keys were removed
+    assert (value_data := node.data["values"][value_idx]) is not None
+    assert value_data["metadata"]
+    assert value_data["value"] == 1
```

Comment:
Maybe assert before the event the original value of the things we assert as updated after the event?